src/client/eppo-client.ts

```diff
@@ -72,42 +44,30 @@ export interface IEppoClient {
   getStringAssignment(
     subjectKey: string,
     flagKey: string,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    subjectAttributes?: Record<string, any>,
-    assignmentHooks?: IAssignmentHooks,
-  ): string | null;
+    defaultValue: string,
+    subjectAttributes?: Record<string, AttributeType>,
+  ): string;
 
   getBoolAssignment(
     subjectKey: string,
     flagKey: string,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    subjectAttributes?: Record<string, any>,
-    assignmentHooks?: IAssignmentHooks,
-  ): boolean | null;
+    defaultValue: boolean,
+    subjectAttributes?: Record<string, AttributeType>,
+  ): boolean;
 
   getNumericAssignment(
     subjectKey: string,
     flagKey: string,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    subjectAttributes?: Record<string, any>,
-    assignmentHooks?: IAssignmentHooks,
-  ): number | null;
-
-  getJSONStringAssignment(
-    subjectKey: string,
-    flagKey: string,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    subjectAttributes?: Record<string, any>,
-    assignmentHooks?: IAssignmentHooks,
-  ): string | null;
+    defaultValue: number,
+    subjectAttributes?: Record<string, AttributeType>,
+  ): number;
```

Comment:
I think we want to have `getIntegerAssignment()` in the interface too. Super annoying keeping these in sync ideally we ax the interface, but that may be hard.

Reply:
Good catch!